src/core/plugin/luapi_application.h

```diff
@@ -811,6 +812,105 @@ static int applib_addStrokes(lua_State* L) {
     return 0;
 }
 
+/**
+ * Puts a Lua Table of the coordinates of the points of all strokes from the selected layer/selection tool onto the
+ * stack
+ * TODO document the returned table and the parameters
+ */
+static int applib_getStrokes(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    std::string type = luaL_checkstring(L, 1);
+    std::vector<Element*> elements = {};
+    Control* control = plugin->getControl();
+
+    if (type == "Layer") {
+        auto sel = control->getWindow()->getXournal()->getSelection();
+        if (sel) {
+            control->clearSelection();  // otherwise strokes in the selection won't be recognized
+        }
+        elements = control->getCurrentPage()->getSelectedLayer()->getElements();
+    } else if (type == "Selection") {
+        auto sel = control->getWindow()->getXournal()->getSelection();
+        if (sel) {
+            elements = sel->getElements();
+        } else {
+            g_warning("There is no selection! ");
+            return 0;
+        }
+    } else {
+        g_warning("Unknown argument: %s", type.c_str());
+        return 0;
+    }
+
+    lua_newtable(L);  // create table of the elements
+    int currStrokeNo = 0;
+    int currPointNo = 0;
+
+    for (Element* e: elements) {
+        if (e->getType() == ELEMENT_STROKE) {
+            auto* s = dynamic_cast<Stroke*>(e);
```

Comment:
```suggestion
            auto* s = static_cast<Stroke*>(e);
```
looks preferable, since we have already checked that it is a `Stroke`